src/main/java/org/opentripplanner/middleware/auth/RequestingUser.java

```diff
@@ -25,37 +31,54 @@
      * Constructor is only used for creating a test user. If an Auth0 user id is provided check persistence for matching
      * user else create default user.
      */
-    private RequestingUser(String auth0UserId) {
-        if (auth0UserId == null) {
-            this.auth0UserId = "user_id:string";
-            otpUser = new OtpUser();
-            apiUser = new ApiUser();
-            adminUser = new AdminUser();
-        } else {
-            this.auth0UserId = auth0UserId;
-            Bson withAuth0UserId = Filters.eq("auth0UserId", auth0UserId);
-            otpUser = Persistence.otpUsers.getOneFiltered(withAuth0UserId);
-            apiUser = Persistence.apiUsers.getOneFiltered(withAuth0UserId);
-            adminUser = Persistence.adminUsers.getOneFiltered(withAuth0UserId);
-        }
+    private RequestingUser(String auth0UserId, String scope) {
+        this.auth0UserId = Objects.requireNonNullElse(auth0UserId, "user_id:string");
+        defineUserFromScope(scope, true);
     }
 
     /**
-     * Create a user profile from the request's JSON web token. Check persistence for stored user.
+     * Create a user profile from the request's JSON web token.
      */
     public RequestingUser(DecodedJWT jwt) {
         this.auth0UserId = jwt.getClaim("sub").asString();
+        String scope = jwt.getClaim("scope").asString();
+        defineUserFromScope(scope, false);
+    }
+
+    /**
+     * Check persistence for stored user restricted by the scope value provided in the scope claim. It is expected that
+     * the scope claim will include only one scope item i.e. 'otp-user', 'api-user' or 'admin-user'. if testing, create
+     * a new user else attempt to get a matching user from DB.
+     */
+    private void defineUserFromScope(String scope, boolean testing) {
         Bson withAuth0UserId = Filters.eq("auth0UserId", auth0UserId);
-        otpUser = Persistence.otpUsers.getOneFiltered(withAuth0UserId);
-        apiUser = Persistence.apiUsers.getOneFiltered(withAuth0UserId);
-        adminUser = Persistence.adminUsers.getOneFiltered(withAuth0UserId);
+        if (scope == null || scope.isEmpty()) {
+            LOG.error("Required scope claim unavailable");
+            return;
+        }
+
+        if (scope.contains(OtpUser.SCOPE)) {
+            otpUser = (testing)
```

Comment:
There might be an opportunity to consolidate these user fields into a single `AbstractUser user` field, but I'm not sure about that approach. Just pointing it out in case we want to add a TODO.

Reply:
In the admin UI, a user could be both an `AdminUser` and an `ApiUser`, how would that work out for them?

Reply:
@binh-dam-ibigroup it will be the responsibility of the calling service in this case the admin UI to define the scope of the calling user. This would either be `admin-user` or `api-user`. It would be good to test this from a third party source if possible? To date, I've only tested via unit tests with `Auth0Users#getApiUserAuth0TokenResponse`.

Reply:
Because both kinds of users can log in from the same page, I would guess the admin UI should set both scopes. As a result, I would recommend to split the `else if` into separate `if` blocks.

Reply:
@binh-dam-ibigroup, the `else if` blocks were added so that we could avoid cases were a requester is identified as more than one user for a given request. Maybe we can think of a way to keep these in place for the admin UI. How about this:
```java
        Bson withAuth0UserId = Filters.eq("auth0UserId", auth0UserId);
        if (scope == null || scope.isEmpty()) {
            LOG.error("Required scope claim unavailable");
            return;
        }
        // Define only a single user according to the scope. Note: there is an edge case where a user logging in from 
        // the OTP Admin Dashboard may be both an AdminUser and ApiUser, but this code block will force their identity
        // as an AdminUser.
        // TODO: Consider consolidating the user scope fields into a single AbstractUser user field.
        if (scope.contains(OtpUser.AUTH0_SCOPE)) {
            otpUser = (testing)
                ? new OtpUser()
                : Persistence.otpUsers.getOneFiltered(withAuth0UserId);
            return;
        }
        if (scope.contains(AdminUser.AUTH0_SCOPE)) {
            adminUser = (testing)
                ? new AdminUser()
                : Persistence.adminUsers.getOneFiltered(withAuth0UserId);
            return;
        }
        if (scope.contains(ApiUser.AUTH0_SCOPE)) {
            apiUser = (testing)
                ? new ApiUser()
                : Persistence.apiUsers.getOneFiltered(withAuth0UserId);
            return;
        }
        LOG.error("No user type for scope {} is available", scope);
```

Reply:
I was thinking something like that. There is a side effect, though, with this change. If you log in as an admin, you will be prompted to create an API user profile, even though you have created one already.

Reply:
I don't understand why that would be the case. If that's due to something weird in the OTP Admin UI, we should just fix it.

Reply:
It is happening in MOD UI too.

Reply:
Disregard my comments regarding admin-UI and MOD-UI, I needed to add the scopes to those.

Reply:
This is the middleground I would use as a fix:
```java
        if (scope.contains(OtpUser.AUTH0_SCOPE)) {
            otpUser = (testing)
                ? new OtpUser()
                : Persistence.otpUsers.getOneFiltered(withAuth0UserId);
            return;
        }
        if (scope.contains(AdminUser.AUTH0_SCOPE)) {
            adminUser = (testing)
                ? new AdminUser()
                : Persistence.adminUsers.getOneFiltered(withAuth0UserId);
        }
        if (scope.contains(ApiUser.AUTH0_SCOPE)) {
            apiUser = (testing)
                ? new ApiUser()
                : Persistence.apiUsers.getOneFiltered(withAuth0UserId);
        }
        LOG.error(...)
```

Reply:
I have updated, but this goes against the idea of limiting request to a single user type.

Reply:
Apologies, after discussion with @landonreed, please use his suggestion instead. The decision is to not support both `AdminUser` and `ApiUser` on the same Auth0 id.

Reply:
Let's use this:
```java
// Define only a single user according to the scope. Note: there is an edge case where a user logging in from
// the OTP Admin Dashboard may be both an AdminUser and ApiUser, but this code block will force their identity
// as an AdminUser.
// TODO: Consider consolidating the user scope fields into a single AbstractUser user field.
if (scope.contains(OtpUser.AUTH0_SCOPE)) {
    otpUser = (testing)
        ? new OtpUser()
        : Persistence.otpUsers.getOneFiltered(withAuth0UserId);
    return;
}
if (scope.contains(AdminUser.AUTH0_SCOPE)) {
    adminUser = (testing)
        ? new AdminUser()
        : Persistence.adminUsers.getOneFiltered(withAuth0UserId);
    // Only return at this point if an AdminUser is defined. If not, fall back on defining an ApiUser. If we
    // returned unconditionally, ApiUsers that log in via OTP Admin Dashboard would not be defined below (the
    // final conditional block would never execute.
    if (adminUser != null) return;
}
if (scope.contains(ApiUser.AUTH0_SCOPE)) {
    apiUser = (testing)
        ? new ApiUser()
        : Persistence.apiUsers.getOneFiltered(withAuth0UserId);
    return;
}
LOG.error("No user type for scope {} is available", scope);
```

Reply:
Updated.